### ISSUE TITLE
Format facet for Number types is not fully specified

### ISSUE THREAD

**Issue by LorenaSoledad** [raml-org/raml-spec#535]
From the _format_ facet description for Number types:

> 
The format of the value. The value MUST be one of the following: int32, int64, int, long, float, double, int16, int8

The spec defines these values for the _format_ field, but It is never specified the exact range of values accepted in each one. Specifically int32, int64, int, int16, int8

https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#number

For example, commonly, an "int" value is 32 bit, so what is the difference with "int32"?

**Comment by freddrake** [comment]
I find the term "format" and it's description there very confusing.  If this is really about specifying the range of the value, then "format" makes little sense, and minimum/maximum should be used instead.

The impression is that "format" is about the underlying representation of the value in an implementation, which I don't think makes sense in RAML.  Specifying float/double (if those mean what they do in C) does have implications on the set of allowed values that can't be expressed via minimum/maximum constraints, but this doesn't allow alternate numeric implementations that could represent whatever is present in the serialized data.

**Comment by sichvoge** [comment]
I think that goes even further since `integer` inherits its facets from `number`, so currently you can specify the following:

```yaml
types:
  Weight:
    type: integer
    format: float
```

I would say that's kinda wrong isn't it?

**Comment by freddrake** [comment]
@sichvoge Whether that's wrong depends on what "format" really means.  It's acceptable to format an integer as a floating-point number in a serialization; every digit right of the decimal will be zero. Allowable serializations are a matter of the serialization syntax (JSON, XML, whatever); the type should be about the allowed values.

Thinking more about this, there probably is a place for "float" and "double", because those affect the possible values.  I'm sure XSD schemas allow all manner of flavors (I've not dealt with numerics using XSD), but for JSON-encodable types, "number" and it's limitations may be as good as we're going to get.  If greater range is required, JSON APIs won't be using numeric types anyway.  Many parts of numeric towers applications care about just aren't representable in JSON.

(No, I'm not suggesting we need to deal with all that. The RAML spec should simply be clear about the values supportable by types defined in RAML.)

**Comment by sichvoge** [comment]
An update for ranges and mappings for `int` and `long` formats.

```
int8
int16
int32 == int
int64 == long
```

And ranges:

```
int8 -- (-128 to 127)

int16 -- (-32,768 to +32,767)

int32 -- (-2,147,483,648 to +2,147,483,647)

int64 -- (-9,223,372,036,854,775,808 to +9,223,372,036,854,775,807)
```

It should be clarified in the spec!

**Comment by PinoEire** [comment]
i see that this is still open... this is one of the main reasons making very difficult to implement a .Net plugin capable of implementing APIs from a RAML project, not the only issue, but the main.

**Comment by sichvoge** [comment]
Yes, that's still open and pending to go into the spec. Hope to get this done as soon as we have the right guys from the workgroup giving their "LGTM".

**Comment by zeroum** [comment]
The spec also does not say, nor it delegates to another RFC, if the numbers are allowed be sent raw (123.45), and/or string-wrapped ("123.45"), or if they can be formatted in a locale-specific way(decimal and "thousands" separators), or if the digits themselves are defined within [0-9](as obvious it may seem... but why not binary, octals, hexadecimals?) or if that is an implementation, mime-type specific concern.

**Comment by sichvoge** [comment]
You mean when you have to validate an instance (through payload) against a type definition?

**Comment by alvassin** [comment]
@sichvoge so, i need to check a length of float|int to match format? Just question of length?

One more question about numbers & integers. In [number documentation](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#number) it is said, that they are applicable only to parameters of type number or integer. Can you clarify that please?
